src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+                gencodeFuncotations.add( createIgrFuncotation(altAllele) );
+
+            } else if (GencodeGtfExonFeature.class.isAssignableFrom(containingSubfeature.getClass())) {
+
+                // We have a coding region variant
+                gencodeFuncotations.add( createCodingRegionFuncotation(variant, altAllele, gtfFeature, reference, transcript, (GencodeGtfExonFeature) containingSubfeature) );
+
+            } else if (GencodeGtfUTRFeature.class.isAssignableFrom(containingSubfeature.getClass())) {
+
+                // We have a UTR variant
+                gencodeFuncotations.add( createUtrFuncotation(variant, altAllele, reference, gtfFeature, transcript, (GencodeGtfUTRFeature) containingSubfeature) );
+
+            } else if (GencodeGtfTranscriptFeature.class.isAssignableFrom(containingSubfeature.getClass())) {
+
+                // We have an intron variant
+                gencodeFuncotations.add( createIntronFuncotation(variant, altAllele, gtfFeature, transcript) );
+
+            } else {
+
+                // Uh-oh!  Problemz.
+                throw new GATKException("Unable to determine type of variant-containing subfeature: " + containingSubfeature.getClass().getName());
+            }
+        }
+
+        return gencodeFuncotations;
+    }
+
+    /**
+     * Create a {@link GencodeFuncotation} for a {@code variant} that occurs in a coding region in a {@code transcript}.
+     * @param variant The {@link VariantContext} for which to create a {@link GencodeFuncotation}.
+     * @param altAllele The {@link Allele} in the given {@code variant} for which to create a {@link GencodeFuncotation}.
+     * @param gtfFeature The {@link GencodeGtfGeneFeature} in which the given {@code variant} occurs.
+     * @param reference The {@link ReferenceContext} for the current data set.
+     * @param transcript The {@link GencodeGtfTranscriptFeature} in which the given {@code variant} occurs.
+     * @param exon The {@link GencodeGtfExonFeature} in which the given {@code variant} occurs.
+     * @return A {@link GencodeFuncotation} containing information about the given {@code variant} given the corresponding {@code exon}.
+     */
+    private GencodeFuncotation createCodingRegionFuncotation(final VariantContext variant,
+                                                             final Allele altAllele,
+                                                             final GencodeGtfGeneFeature gtfFeature,
+                                                             final ReferenceContext reference,
+                                                             final GencodeGtfTranscriptFeature transcript,
+                                                             final GencodeGtfExonFeature exon) {
+
+        // Setup the "trivial" fields of the gencodeFuncotation:
+        final GencodeFuncotation gencodeFuncotation = createGencodeFuncotationWithTrivialFieldsPopulated(variant, altAllele, gtfFeature, transcript);
```

Comment:
Perhaps a builder pattern would be more appropriate.  Since the gencodeFuncotation is in a bad state for a good chunk of this code.

Reply:
This would also allow you (or future developers) to validate that any GencodeFuncotation is ready for use and has not been created improperly.

Reply:
Yeah - it's about 50% the way there to a Builder pattern already.

I'll update it to have a builder that can populate the fields.